test/controller/FakeSoundController.vala

```diff
@@ -0,0 +1,262 @@
+/*
+ * brickman -- Brick Manager for LEGO MINDSTORMS EV3/ev3dev
+ *
+ * Copyright (C) 2016 Kaelin Laundry <wasabifan@outlook.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+/* FakeSoundController.vala - Fake sound controller for testing */
+
+using Ev3devKit;
+using Ev3devKit.Ui;
+
+namespace BrickManager {
+    public class FakeSoundController : Object, IBrickManagerModule {
+        private const int VOLUME_STEP = 10;
+
+        MixerElementSelectorWindow mixer_select_window;
+        MixerElementVolumeWindow volume_window;
+
+        public string display_name { get { return "Sound"; } }
+
+        public FakeSoundController (Gtk.Builder builder) {
+            // Initialize windows that the controller needs
+            mixer_select_window = new MixerElementSelectorWindow ();
+            volume_window = new MixerElementVolumeWindow ();
+
+            // Register for callback so that we can focus on the correct control panel tab
+            // the first time either window is invoked
+            var control_panel_notebook = builder.get_object ("control-panel-notebook") as Gtk.Notebook;
+            mixer_select_window.shown.connect (() =>
+                control_panel_notebook.page = (int)ControlPanel.Tab.SOUND);
+            volume_window.shown.connect (() =>
+                control_panel_notebook.page = (int)ControlPanel.Tab.SOUND);
+            
+            // Initialize items in brickman for all current elements
+            var mixer_elems_liststore = builder.get_object ("mixer-elements-liststore") as Gtk.ListStore;
+            mixer_elems_liststore.foreach ((model, path, iter) => {
+                update_fake_element_from_liststore (iter, mixer_elems_liststore);
+                return false;
+            });
+
+            // Propagate changes from liststore to views
+            mixer_elems_liststore.row_changed.connect ((path, iter) => {
```

Comment:
reference cycle on `mixer_elems_liststore`, although it probably doesn't matter if we never free `mixer_elems_liststore`.